texar/torch/modules/tokenizers/pretrained_gpt2_tokenizer.py

```diff
@@ -0,0 +1,256 @@
+
+    def forward(self, inputs, task, **kwargs):
+        if task == 'text-to-token':
+            return self.tokenize(inputs)
+        elif task == 'token-to-text':
+            return self.convert_tokens_to_string(inputs)
+        elif task == 'text-to-id':
+            return self.encode(inputs)
+        elif task == 'id-to-text':
+            skip_special_tokens = kwargs.get('skip_special_tokens', False)
+            clean_up_tokenization_spaces = kwargs.get(
+                'clean_up_tokenization_spaces', True)
+            return self.decode(
+                inputs, skip_special_tokens=skip_special_tokens,
+                clean_up_tokenization_spaces=clean_up_tokenization_spaces)
+        elif task == 'token-to-id':
+            return self.convert_tokens_to_ids(inputs)
+        elif task == 'id-to-token':
+            skip_special_tokens = kwargs.get('skip_special_tokens', False)
+            return self.convert_ids_to_tokens(
+                inputs, skip_special_tokens=skip_special_tokens)
+        else:
+            raise ValueError("Unrecognized job type.")
+
+    def _tokenize(self, text: str) -> List[str]:  # type: ignore
+        r"""Tokenize a string. """
+        bpe_tokens: List[str] = []
+        for token in re.findall(self.pat, text):
+            token = ''.join(self.byte_encoder[b] for b in token.encode('utf-8'))
+            bpe_tokens.extend(
+                bpe_token for bpe_token in self.bpe(token).split(' '))
+        return bpe_tokens
+
+    def save_vocabulary(self, save_directory: str) -> Tuple[str, str]:
+        r"""Save the tokenizer vocabulary and merge files to a directory."""
+        if not os.path.isdir(save_directory):
+            raise ValueError("Vocabulary path ({}) should be a "
+                             "directory".format(save_directory))
+
+        vocab_file = os.path.join(save_directory,
+                                  self._VOCAB_FILE_NAMES['vocab_file'])
+        merge_file = os.path.join(save_directory,
+                                  self._VOCAB_FILE_NAMES['merges_file'])
+
+        with open(vocab_file, 'w', encoding='utf-8') as f:
+            f.write(json.dumps(self.encoder, ensure_ascii=False))
+
+        index = 0
+        with open(merge_file, "w", encoding="utf-8") as writer:
+            writer.write(u'#version: 0.2\n')
+            for bpe_tokens, token_index in sorted(self.bpe_ranks.items(),
+                                                  key=lambda kv: kv[1]):
+                if index != token_index:
+                    print("Saving vocabulary to {}: BPE merge indices are "
+                          "not consecutive. Please check that the tokenizer "
+                          "is not corrupted!".format(merge_file))
+                    index = token_index
+                writer.write(' '.join(bpe_tokens) + u'\n')
+                index += 1
+
+        return (vocab_file, merge_file)
+
+    def bpe(self, token: str) -> str:
```

Comment:
if not exposed to users, make it private. Otherwise it will be shown in the online doc